File: GasStation/GasStation.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        vector<int>::size_type Stations = gas.size();
        int total = 0, sum = 0, start = 0;
        for(vector<int>::size_type i = 0; i < Stations; i++) {
        	total += gas[i] - cost[i];
        	if(sum < 0) {
        		sum = gas[i] - cost[i];
        		start = i;
        	}
        	else
        		sum += gas[i] - cost[i];
        }
        if(total < 0)
        	return -1;
        else
        	return start;
    }
};

File: GasStation/GasStation.py
class Solution(object):
    def canCompleteCircuit(self, gas, cost):
        """
        :type gas: List[int]
        :type cost: List[int]
        :rtype: int
        """
        total = 0
        tmp_sum = 0
        start = 0
        for i in range(len(gas)):
            total += gas[i] - cost[i]
            if tmp_sum < 0:
                tmp_sum = gas[i] - cost[i]
                start = i
            else:
                tmp_sum += gas[i] - cost[i]
        if total >= 0:
            return start
        else:
            return -1